echopype/tests/mask/test_mask.py

```diff
@@ -219,6 +275,133 @@ def test_frequency_differencing(n: int, n_chan_freq: int,
     assert out.name == "mask"
 
 
+@pytest.mark.parametrize(
+    ("n", "n_chan", "var_name", "mask", "mask_file", "fill_value", "is_delayed", "var_masked_truth"),
+    [
+        (2, 1, "var1", np.identity(2), None, np.nan, False, np.array([[1, np.nan], [np.nan, 1]])),
+        (2, 1, "var1", np.identity(2), None, 2.0, False, np.array([[1, 2.0], [2.0, 1]])),
+        (2, 1, "var1", np.identity(2), None, np.array([[[np.nan, np.nan], [np.nan, np.nan]]]),
+         False, np.array([[1, np.nan], [np.nan, 1]])),
+        (2, 1, "var1", np.identity(2), None, xr.DataArray(data=np.array([[[np.nan, np.nan], [np.nan, np.nan]]]),
+                                                          coords={"channel": ["chan1"],
+                                                                  "ping_time": [0, 1],
+                                                                  "range_sample": [0, 1]}),
+         False, np.array([[1, np.nan], [np.nan, 1]])),
+        (2, 1, "var1", [np.identity(2), np.array([[0, 1], [0, 1]])], [None, None], 2.0,
+         False, np.array([[2.0, 2.0], [2.0, 1]])),
+        (2, 1, "var1", np.identity(2), None, 2.0, True, np.array([[1, 2.0], [2.0, 1]])),
+        (2, 1, "var1", np.identity(2), "test.zarr", 2.0, True, np.array([[1, 2.0], [2.0, 1]])),
+        (2, 1, "var1", [np.identity(2), np.array([[0, 1], [0, 1]])], ["test0.zarr", "test1.zarr"], 2.0,
+         False, np.array([[2.0, 2.0], [2.0, 1]])),
+        (2, 1, "var1", [np.identity(2), np.array([[0, 1], [0, 1]])], ["test0.zarr", None], 2.0,
+         False, np.array([[2.0, 2.0], [2.0, 1]])),
+    ],
+    ids=["single_mask_default_fill", "single_mask_float_fill", "single_mask_np_array_fill",
+         "single_mask_DataArray_fill", "list_mask_all_np", "single_mask_ds_delayed",
+         "single_mask_as_path", "list_mask_all_path", "list_mask_some_path"]
+)
+def test_apply_mask(n: int, n_chan: int, var_name: str,
+                    mask: Union[np.ndarray, List[np.ndarray]],
+                    mask_file: Optional[Union[str, List[str]]],
+                    fill_value: Union[int, float, np.ndarray, xr.DataArray],
+                    is_delayed: bool, var_masked_truth: np.ndarray):
```

Comment:
I think what test contains right now is more of the integrated test flavor, and some of those can be split out to unit test the mask validation and collection function.

My suggestion is create/split things into the following:
- unit tests for `validate_and_collect_mask_input`: 
  - test allowable types of mask sources
  - test the allowable forms of mask (list or single)
- unit tests for `_check_var_name_fill_value`
  - test whether `var_name` is in `source_ds`
  - test whether `fill_value` is of the same shape as `source_ds["var_name"]`
- integration tests that actually applies the tests (you have these already)